core/cog-directory-files-handler.c

```diff
@@ -187,6 +187,24 @@ cog_directory_files_handler_run (CogRequestHandler      *request_handler,
         ++path;
 
     g_autoptr(GFile) file = g_file_get_child (base_path, path);
+
+    /*
+     * Check whether the resolved path is contained inside base_path,
+     * to prevent URIs with ".." components from accessing resources
+     * outside of base_path. The g_file_get_relative_path() method
+     * returns NULL when the file path is NOT descendant of base_path.
+     */
+    g_autofree char *relative_path = g_file_get_relative_path (base_path, file);
+    if (!relative_path) {
```

Comment:
This introduces a regression: the check does not cover the case when `relative_path` and `base_path` end up being the same path, so if the path is empty an error is returned instead of checking whether an `index.html` file is present at the base path. I'll post a PR with a fix momentarily.

Reply:
Fix in #266